src/main/java/org/reactome/addlinks/AddLinks.java

```diff
@@ -513,20 +517,88 @@ else if (fileProcessorName.get() instanceof List<?>)
 						}
 					}
 				}
-
+				if (refCreator instanceof NCBIGeneBasedReferenceCreator || refCreator instanceof ENSMappedIdentifiersReferenceCreator)
+				{
+					for (EntrezGeneBasedReferenceCreator subCreator : ((NCBIGeneBasedReferenceCreator)refCreator).getSubCreators())
+					{
+						String line = checkLinksForRefCreator(subCreator);
+						Files.write(Paths.get(linkCheckReportName), line.getBytes(), StandardOpenOption.APPEND);
+					}
+				}
+				// Now check the links for the reference creator.
+				String line = checkLinksForRefCreator(refCreator);
+				Files.write(Paths.get(linkCheckReportName), line.getBytes(), StandardOpenOption.APPEND);
 			}
 			// There is a separate list of reference creators to create UniProt references.
 			else if (this.uniprotReferenceCreators.containsKey(refCreatorName))
 			{
 				UPMappedIdentifiersReferenceCreator refCreator = this.uniprotReferenceCreators.get(refCreatorName);
-				if (refCreator instanceof NCBIGeneBasedReferenceCreator)
-				{
-					((NCBIGeneBasedReferenceCreator) refCreator).setCTDGenes( (Map<String, String>) dbMappings.get(CTD_PROCESSOR) );
-				}
+				refCreator.setCTDGenes( (Map<String, String>) dbMappings.get(CTD_PROCESSOR) );
+
 				sourceReferences = this.getIdentifiersList(refCreator.getSourceRefDB(), refCreator.getClassReferringToRefName());
 				refCreator.createIdentifiers(personID, (Map<String, Map<String, List<String>>>) dbMappings.get(fileProcessorName.get()), sourceReferences);
+				// Now check the links for the reference creator (and sub-creators).
+				for (EntrezGeneBasedReferenceCreator subCreator : refCreator.getSubCreators())
+				{
+					String line = checkLinksForRefCreator(subCreator);
+					Files.write(Paths.get(linkCheckReportName), line.getBytes(), StandardOpenOption.APPEND);
+				}
+				String line = checkLinksForRefCreator(refCreator);
+				Files.write(Paths.get(linkCheckReportName), line.getBytes(), StandardOpenOption.APPEND);
+			}
+		}
+	}
+
+	/**
+	 * @param refCreator
+	 * @return
+	 */
+	private String checkLinksForRefCreator(BatchReferenceCreator<?> refCreator)
+	{
+		String targetRefDB = refCreator.getTargetRefDB();
+		String line = "";
+		if (!targetRefDB.toUpperCase().contains(KEGG) && !targetRefDB.toUpperCase().contains(ENSEMBL))
+		{
+			Optional<GKInstance> refDBInst = LinksToCheckCache.getCache().keySet().stream()
+																.filter( inst -> ((inst.getDisplayName().equals(targetRefDB) && LinksToCheckCache.getRefDBsToCheck().contains(targetRefDB))))
+																.findFirst();
+			if (refDBInst.isPresent())
+			{
+				line = checkLinksForRefDB(new LinkCheckManager(), refDBInst.get());
 			}
+			else
+			{
+				line = targetRefDB + "\tN/A\tN/A\n";
+			}
+			return line;
```

Comment:
Could this be reduced to
```
String line = refDBInst.isPresent() ? 
    checkLinksForRefDB(new LinkCheckManager(), refDBInst.get()) : 
    targetRefDB + "\tN/A\tN/A\n"`;
```

Reply:
Probably, but I don't like ternaries that much. I find this suggested replacement a little hard to read. I am usually OK with ternaries when the expressions are much shorter.

Reply:
I mean... it could be reduced even further to 
```Java
return refDBInst.isPresent() ? 
          checkLinksForRefDB(new LinkCheckManager(), refDBInst.get()) : 
          targetRefDB + "\tN/A\tN/A\n"
```

But I find this _waaaaay_ too terse.

Reply:
Okay